### PR TITLE
Q3 taxonomy misc changes

### PR DESCRIPTION
**Pull request by cbaines**
As part of the Taxonomy team work in Q3, I successfully generated an inverse similarity spreadsheet for the Coming to the UK theme with this script [1].

I've attempted to make this someone Python 3 compatible, but I also haven't checked yet if these changes break Python 2 compatibility. I think there was only one issue with the dependencies, and that is the pattern library, that doesn't have a release that works with Python 3 yet ( https://github.com/clips/pattern/issues/62 ).

I also ended up reverting some of the work on multithreading, as this looked like it wouldn't work, and didn't work when I tested it.

1: https://trello.com/c/ogDgie3I/92-generate-inverse-similarity-sample-of-cttuk-content

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,38 +16,18 @@ Based on a review of the Education themed content, this algorithm performed sign
 First, you'll need to install some dependencies. It's a python thing, so:
 
 ```
-$ git clone git@github.com:alphagov/ordering-documents-by-inverse-similarity.git
-$ cd ordering-documents-by-inverse-similarity
-$ virtualenv .venv
-$ source .venv/bin/activate
-$ pip install -r requirements.txt
+$ git clone git@github.com:alphagov/govuk-inverse-similarity.git
+$ govuk-inverse-similarity
```
Comment (tomsabin) [inline README.md]: This line isn't needed - the file doesn't exist so I'm not sure if this was intended on being some kind of setup script?

Comment (cbaines) [reply README.md]: Ah, I've missed the `cd` off of the start. I'll update. Good spot :)

**Comment by tomsabin** [comment]
I've tried installing the requirements with Python 3 and I've got as far as the Pattern module. 

<details>
  <summary><code>pip3 install -r requirements.txt</code> output</summary>

```bash
Collecting Pattern==2.6 (from -r requirements.txt (line 9))
  Downloading pattern-2.6.zip (24.6MB)
    100% |████████████████████████████████| 24.6MB 59kB/s
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/bt/94l56_lx3jv1ht7hs92xbqz40000gn/T/pip-build-uj2gl2pv/Pattern/setup.py", line 40
        print n
              ^
    SyntaxError: Missing parentheses in call to 'print'. Did you mean print(int n)?

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/bt/94l56_lx3jv1ht7hs92xbqz40000gn/T/pip-build-uj2gl2pv/Pattern/
```

</details>

<br>

According to the [installation notes](https://github.com/clips/pattern): "The Python 3 version is currently only available on the development branch."


I haven't used Python enough to know the best practice to manage the two versions, but is there a nice workaround to have a compatible `requirements.txt` for both versions?

In the meantime, I'll try with Python 2.

**Comment by cbaines** [comment]
> I haven't used Python enough to know the best practice to manage the two versions, but is there a nice workaround to have a compatible requirements.txt for both versions?

We could possibly do something with the requirements.txt. If you wanted to try with Python 3, you could try installing with: 

```
pip3 install git+https://github.com/clips/pattern.git@development
```

**Comment by tomsabin** [comment]
This could possibly work in the requirements file after reading [this requirement-specifiers documentation](http://pip.readthedocs.io/en/stable/reference/pip_install/#requirement-specifiers):

```diff
stop_words
gensim==0.13.1
- Pattern==2.6
+ -e git+git@github.com:clips/pattern.git@development#egg=pattern-development ; python_version > '2.7'
+ Pattern==2.6 ; python_version <= '2.7'
ipdb==0.10.1
Scrapy==1.2.1

```

Edit: updated to use Python 3 dependencies only in https://github.com/alphagov/govuk-inverse-similarity/pull/2/commits/04b49346b50cf559f6d5231902a52f5c783347e5

**Comment by tomsabin** [comment]
Now the multithreading has been removed, it's probably worth removing the subsection in "Advanced configuration" in the readme too:

https://github.com/alphagov/govuk-inverse-similarity/tree/q3-taxonomy-misc-changes#number-of-cores

> Number of Cores
> 
> This sets the number of processes used to train the LDA Model, and to map topics to page content. It defaults to 1. If you've got spare CPU cycles to throw at it, set this to the number of CPU cores you have available, minus 1.

Edit: removed from README in https://github.com/alphagov/govuk-inverse-similarity/pull/2/commits/fb1249819460fc050473939d99869d6fb46b8654

**Review by tomsabin — APPROVED** [review]

**Pull request merged**